src/Microsoft.Health.Dicom.Core/Features/Update/UpdateInstanceService.cs

```diff
@@ -187,11 +193,12 @@ private async Task UpdateDatasetInFileAsync(long newFileIdentifier, DicomDataset
         using MemoryStream resultStream = _recyclableMemoryStreamManager.GetStream(tag: DestTag);
         await dicomFile.SaveAsync(resultStream);
 
-        await _fileStore.UpdateFileBlockAsync(newFileIdentifier, block.Key, resultStream, cancellationToken);
+        FileProperties fileProperties = await _fileStore.UpdateFileBlockAsync(newFileIdentifier, partition, block.Key, resultStream, cancellationToken);
 
         stopwatch.Stop();
 
         _logger.LogInformation("Updating new file {NewFileIdentifier} completed successfully. {TotalTimeTakenInMs} ms", newFileIdentifier, stopwatch.ElapsedMilliseconds);
+        return _externalStoreEnabled ? fileProperties : null;
```

Comment:
What is this for? Because we don't want to record file properties for non-external scenarios? Where else could this determination be made?